srtcore/core.cpp

```diff
@@ -1791,7 +1793,10 @@ bool CUDT::createSrtHandshake(ref_t<CPacket> r_pkt, ref_t<CHandShake> r_hs,
         size_t aligned_bytesize = wordsize*4;
 
         memset(p+offset, 0, aligned_bytesize);
+
         memcpy(p+offset, sm.data(), sm.size());
+        // Preswap to little endian (in place due to possible padding zeros)
+        HtoILA((uint32_t*)(p+offset), (uint32_t*)(p+offset), wordsize);
```

Comment:
@ethouris How much effort can it take to do copy & swap?
`HtoILACopy((uint32_t*)(p+offset), (uint32_t*)(sm.data()), sm.size());`.
The padding zeros can be detected based on the `size != wordsize`.
What do you think?

Reply:
It's possible, but makes things complicated and more error prone, while it's not performance critical (it's a handshake, so done only once per connection, and the biggest string allowed here is of 512 bytes).